spec/views/admin_general/_to_do_list.html.erb_spec.rb

```diff
@@ -21,10 +21,9 @@ def render_errors(items)
       end
 
       it 'shows "None given" if there is no user message' do
-        original_params = request.last_event.params
-        allow(request.last_event).
-          to receive(:params).
-            and_return(params.delete_if { |key| key == :message } )
+        alt_params = request.last_event.params.delete_if { |k| k == :message }
+        allow_any_instance_of(InfoRequestEvent).
+          to receive(:params).and_return(alt_params)
```

Comment:
This change had me confused for some time. It certainly works but I think rewriting we can make it better and remove the call `allow_any_instance_of` which always strikes me as a code smell. What do you think of:

```diff
diff --git a/spec/factories/info_requests.rb b/spec/factories/info_requests.rb
index 279f5c41b..6505f7e4c 100644
--- a/spec/factories/info_requests.rb
+++ b/spec/factories/info_requests.rb
@@ -90,6 +90,18 @@ FactoryBot.define do
         end
       end
 
+      factory :blank_message_request do
+        after(:create) do |info_request, evaluator|
+          info_request.log_event(
+            'status_update',
+            user_id: info_request.user.id,
+            old_described_state: info_request.described_state,
+            described_state: 'error_message',
+            message: '')
+          info_request.set_described_state('error_message')
+        end
+      end
+
       factory :attention_requested_request do
         after(:create) do |info_request, evaluator|
           info_request.log_event('report_request',
diff --git a/spec/views/admin_general/_to_do_list.html.erb_spec.rb b/spec/views/admin_general/_to_do_list.html.erb_spec.rb
index 2c39821fc..4e4371530 100644
--- a/spec/views/admin_general/_to_do_list.html.erb_spec.rb
+++ b/spec/views/admin_general/_to_do_list.html.erb_spec.rb
@@ -20,12 +20,13 @@ describe 'admin_general/_to_do_list.html.erb' do
         expect(rendered).to include('Fix these delivery and other errors')
       end
 
-      it 'shows "None given" if there is no user message' do
-        alt_params = request.last_event.params.delete_if { |k| k == :message }
-        allow_any_instance_of(InfoRequestEvent).
-          to receive(:params).and_return(alt_params)
-        render_errors(items)
-        expect(rendered).to include('None given')
+      context 'without user message' do
+        let(:request) { FactoryBot.create(:blank_message_request) }
+
+        it 'shows "None given"' do
+          render_errors(items)
+          expect(rendered).to include('None given')
+        end
       end
 
       it 'shows the user message when there is one' do
@@ -56,14 +57,15 @@ describe 'admin_general/_to_do_list.html.erb' do
     end
 
     context 'comment reported as requiring admin attention' do
+      let(:request) { FactoryBot.create(:error_message_request) }
       let(:comment) do
         FactoryBot.create(:attention_requested_comment,
                           reason: 'Annotation contains defamatory material',
-                          message: 'Useful info')
+                          message: request.last_event.params[:message],
+                          info_request: request)
       end
 
       let(:items) { [ comment ] }
-      let(:request) { comment.info_request }
       it_behaves_like 'showing requests in an error state'
 
       it 'shows the reason given for the comment being reported' do
```

Reply:
Hmm. I like the simplification but I don't think this is going to work inside a shared example block...

``` diff
-      it 'shows "None given" if there is no user message' do
-        alt_params = request.last_event.params.delete_if { |k| k == :message }
-        allow_any_instance_of(InfoRequestEvent).
-          to receive(:params).and_return(alt_params)
-        render_errors(items)
-        expect(rendered).to include('None given')
+      context 'without user message' do
+        let(:request) { FactoryBot.create(:blank_message_request) }
+
+        it 'shows "None given"' do
+          render_errors(items)
+          expect(rendered).to include('None given')
+        end
       end
```

I've removed the 'None given' check from shared examples and added a new context block for it instead

Reply:
Hmm though I had tested it as a shared example but maybe I didn't. Your suggestion looks like it would be good

Reply:
Gah, missed the all-important `message: ''`  in the factory. Sorry!